### ISSUE TITLE
Crash due to

### ISSUE THREAD

**Issue by GrouchyGaijin** [tokland/youtube-upload#77]
I had this working in the past.  I recently did a fresh install of Ubuntu 14.04 and now when I try to set up the YouTube Uploader I get this error:

Traceback (most recent call last):
  File "/usr/local/bin/youtube-upload", line 9, in <module>
    from youtube_upload import main    
  File "/usr/local/lib/python2.7/dist-packages/youtube_upload/main.py", line 25, in <module>
    import googleapiclient.errors
ImportError: No module named googleapiclient.errors

**Comment by tokland** [comment]
Did you install google-api-python-client? (I have 1.4.2 in my system):

```
$ pip2 list | grep google-api-python-client
google-api-python-client (1.4.2)
```

**Comment by GrouchyGaijin** [comment]
I can't access the machine at the moment, but that may be the problem.  Is that in the repository or is that an add-on that needs to be installed via pip or easy_install? Can you please remind me of the command to install via pip? 

John

**Comment by GrouchyGaijin** [comment]
Thank you!  That solved the problem.  I had to install pip first, but then it was a simple matter of running
pip install --upgrade google-api-python-client

**Comment by tokland** [comment]
Glad it worked!

**Comment by zachwhalen** [comment]
Hi,

I'm getting the same error, but upgrading my google-api-python-client didn't resolve it:

```
$ pip2 list | grep google-api-python-client
google-api-python-client (1.4.2)
```

Here's the full error I'm getting.
```
Traceback (most recent call last):
  File "/usr/local/bin/youtube-upload", line 9, in <module>
    from youtube_upload import main    
  File "/Library/Python/2.7/site-packages/youtube_upload/main.py", line 24, in <module>
    import googleapiclient.errors
ImportError: No module named googleapiclient.errors
```
Any suggestions? I haven't done much with Python before, so it could be something obvious. 

Thanks in advance.
ZW

**Comment by tokland** [comment]
I have the same:

```$ pip2 list | grep google-api-python-client
google-api-python-client (1.4.2)
```

Google only points here for this error, weird. No OsX to test, sorry. Is this file there?

/Library/Python/2.7/site-packages/googleapiclient/errors.py

**Comment by zachwhalen** [comment]
I don't have it there, but I found it in `/usr/local/lib/python2.7/site-packages/googleapiclient/`. Is there some way to point to youtube_uploader to that prefix?

**Comment by zachwhalen** [comment]
Also, if I do like
```
$ python
 import googleapiclient
 import oauth2client
```
... in the terminal, I don't get any errors. So is there something in youtube_upload or possibly in my Python installation that messes with `sys.path`?

**Comment by tokland** [comment]
The script uses the she-bang `/usr/bin/env python`. What's the output of `which python`.

**Comment by zachwhalen** [comment]
I have python at `usr/local/bin/python`, or at least it's there now. But actually I just came here to post that I seem to have resolved my problem (and that solution I found altered the PYTHONPATH variable.)

I first did something heavy-handed and unwise by manually copying all the relevant modules from `/usr/local/lib/python2.7/site-packages` over to `Library/Python/2.7/site-packages`. This cleared some of the errors, but I was still getting crashes from the google api client code. I eventually found something in the Troubleshooting section of Google's Python [documentation for the Tasks API](https://developers.google.com/google-apps/tasks/quickstart/python), where it advises as follows:

> Determine pip's install location with the following command:
> `pip show six | grep "Location:" | cut -d " " -f2`
> Add the following line your you ~/.bashrc file, replacing <pip_install_path> with the value determined above:
> `export PYTHONPATH=$PYTHONPATH:<pip_install_path>`
> Reload your ~/.bashrc file in any open terminal windows using the following command:
> `source ~/.bashrc`

This resolved all the issues I was having, and I'm able to use youtube_upload as advertised.  Thanks for checking!
